src/main/java/com/code_intelligence/jazzer/junit/JUnitLifecycleMethodsInvoker.java

```diff
@@ -0,0 +1,131 @@
+    }
+    ExtensionRegistry extensionRegistry = maybeExtensionRegistry.get();
+
+    // BeforeEachCallback implementations take precedence over @BeforeEach methods. The annotations
+    // are turned into extensions using an internal adapter class, BeforeEachMethodAdapter.
+    // https://junit.org/junit5/docs/current/user-guide/#extensions-execution-order-wrapping-behavior
+    ArrayList<ThrowingRunnable> beforeEachMethods =
+        Stream
+            .<ThrowingRunnable>concat(
+                extensionRegistry.stream(BeforeEachCallback.class)
+                    .map(callback -> () -> callback.beforeEach(extensionContext)),
+                extensionRegistry.stream(BeforeEachMethodAdapter.class)
+                    .map(callback
+                        -> ()
+                            -> callback.invokeBeforeEachMethod(
+                                extensionContext, extensionRegistry)))
+            .collect(toCollection(ArrayList::new));
+
+    ArrayList<ThrowingRunnable> afterEachMethods =
+        Stream
+            .<ThrowingRunnable>concat(
+                extensionRegistry.stream(AfterEachCallback.class)
+                    .map(callback -> () -> callback.afterEach(extensionContext)),
+                extensionRegistry.stream(AfterEachMethodAdapter.class)
+                    .map(callback
+                        -> ()
+                            -> callback.invokeAfterEachMethod(extensionContext, extensionRegistry)))
+            .collect(toCollection(ArrayList::new));
+    // JUnit calls AfterEach methods in reverse order of registration so that the methods registered
+    // first run last.
+    Collections.reverse(afterEachMethods);
+
+    return new JUnitLifecycleMethodsInvoker(
+        Stream.concat(afterEachMethods.stream(), beforeEachMethods.stream())
+            .toArray(ThrowingRunnable[] ::new));
+  }
+
+  private static Optional<ExtensionRegistry> getExtensionRegistryViaHack(
+      ExtensionContext extensionContext) {
+    // Get the private DefaultExecutableInvoker#extensionRegistry field, using the type rather than
+    // the name for slightly better forwards compatibility.
+    return Arrays.stream(DefaultExecutableInvoker.class.getDeclaredFields())
+        .filter(field -> field.getType() == ExtensionRegistry.class)
+        .findFirst()
+        .flatMap(extensionRegistryField -> {
+          DefaultExecutableInvoker invoker =
+              (DefaultExecutableInvoker) extensionContext.getExecutableInvoker();
+          long extensionRegistryFieldOffset =
+              UnsafeProvider.getUnsafe().objectFieldOffset(extensionRegistryField);
+          return Optional.ofNullable((ExtensionRegistry) UnsafeProvider.getUnsafe().getObject(
+              invoker, extensionRegistryFieldOffset));
+        });
+  }
+
+  @Override
+  public void beforeFirstExecution() {}
+
+  @Override
+  public void beforeEachExecution() throws Throwable {
+    if (timesCalledBetweenExecutions++ == 0) {
```

Comment:
This is ultimately more of a libFuzzer than a JUnit problem: `afterEachExecution` is more difficult to implement since the control flow after an execution is not so clear - we can end up reporting a finding, terminating due to some serious problem or returning from libFuzzer. Enumerating all these cases seems error-prone and luckily JUnit doesn't care since it will do the job for us :-)